program/src/logic.rs

```diff
@@ -31,6 +32,40 @@ pub(crate) fn check_rent_exempt(
     }
     Ok(())
 }
+
+pub(crate) fn check_mint(
```

Comment:
I would add a doc comment that we only need to check the mint during initialization of the Solido instance; otherwise people might think this check is missing in other instruction processor functions.